### PR TITLE
Add backend-aware pretrained checkpoints

### PR DESCRIPTION
**Pull request by kellyguo11**
# Description

## Summary

- Add backend-aware pretrained checkpoint names using
  `<task_name>_<physics_backend>_<render_backend>.<extension>`.
- Resolve `--use_pretrained_checkpoint` from the active physics and renderer
  configuration across RL-Games, RSL-RL, SKRL, SB3, benchmark playback, LEAPP
  export, and the H1 demo.
- Extend the checkpoint management tool with a preferred core-task matrix,
  smoke validation, resumable full training, and local collection grouped by
  RL library.
- Select `isaacsim_physx` explicitly for normalized PhysX checkpoint jobs so
  they cannot resolve to OvPhysX.
- Fix the Shadow Handover Newton actuator configuration, add coordinated
  Isaac Sim PhysX scene/asset presets, and correct the G1 MJWarp schedule.

## Motivation

The previous published layout used one generic workflow checkpoint name, so it
could not distinguish policies trained with PhysX versus Newton or RTX versus
the Newton renderer. This made `--use_pretrained_checkpoint` ambiguous for
multi-backend tasks.

The new layout preserves legacy lookups for callers that omit backend names,
while new playback resolves the exact policy matching the configured backends.

The preferred core matrix uses RSL-RL when available, falls back to RL-Games,
and uses SKRL MAPPO for multi-agent tasks. It preserves each task's default
domain preset, excludes Newton Kamino, and collects checkpoints under
`logs/pretrained_checkpoints/<rl_library>/`.

Franka OSC is excluded from the Newton MJWarp matrix because its articulated
dynamics become numerically unstable under OSC effort commands. Its PhysX job
remains included.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation update

## Screenshots

Not applicable.

## Validation

- `uv run isaaclab -f`
- `uv run isaaclab -d`
- `uv run python tools/changelog/cli.py check develop`
- `uv run --extra test python -m pytest source/isaaclab_rl/test/test_pretrained_checkpoint.py source/isaaclab_tasks/test/core/test_handover_env_cfg.py source/isaaclab_tasks/test/core/test_g1_agent_cfg.py` — 12 passed
- `uv run python -m pytest scripts/tools/test/test_train_and_publish_checkpoints.py source/isaaclab_tasks/test/core/test_handover_physics_presets.py` — 3 passed
- Smoke-trained the selected core-task backend combinations for one iteration
  with no failures.
- Verified each task regression test fails against its unfixed configuration
  and passes with the fix.
- Verified `--use_pretrained_checkpoint` loads a backend-named local checkpoint
  in an end-to-end Cartpole playback.
- Completed the requested full-training retry queue, using `isaacsim_physx` for
  PhysX jobs and running the Reorient jobs last.
- Loaded all 69 locally collected checkpoint artifacts with PyTorch and
  verified their backend-aware filenames.

## Checkpoint artifacts

Generated checkpoints and logs are intentionally not included in this pull
request. They are stored locally under
`logs/pretrained_checkpoints/<rl_library>/` for separate publication.

The 20 current matrix entries without collected artifacts consist of 16
explicitly skipped camera combinations and four earlier Newton failures
(Franka cloth lift and rough Cassie, G1, and H1 locomotion). They are not
regressions from this pull request.

## Checklist

- [x] I have read and understood the contribution guidelines
- [x] I have run the pre-commit checks
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fixes and feature work
- [x] I have added a changelog fragment for every touched package
- [x] My name already exists in `CONTRIBUTORS.md`

### CODE REVIEW

**Comment by frlai** [comment]
do the export scripts need this as well to use the pretrained checkpoints?

**Review by StafaH — APPROVED** [review]
